samples/snippets/src/test/java/com/example/dialogflow/CreateConversationProfileTest.java

```diff
@@ -0,0 +1,102 @@
+
+@RunWith(JUnit4.class)
+@SuppressWarnings("checkstyle:abbreviationaswordinname")
+public class CreateConversationProfileTest {
+
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+
+  private static void requireEnvVar(String varName) {
+    assertNotNull(System.getenv(varName));
+  }
+
+  @BeforeClass
+  public static void checkRequirements() {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("GOOGLE_CLOUD_PROJECT");
+  }
+
+  @Test
+  public void testCreateConversationProfileArticleFaq() throws IOException {
+    String conversationProfileDisplayName = UUID.randomUUID().toString();
+    String location = "global";
+    
+    // Create a conversation profile
+    String articleSuggestionKnowledgeBaseId = UUID.randomUUID().toString();
+    String faqKnowledgeBaseId = UUID.randomUUID().toString();
+    ConversationProfile createdConversationProfile = 
+        ConversationProfileManagement.createConversationProfileArticleFaq(
+          PROJECT_ID, 
+          conversationProfileDisplayName, 
+          location,
+          Optional.of(articleSuggestionKnowledgeBaseId), 
+          Optional.of(faqKnowledgeBaseId));
+    assertEquals(conversationProfileDisplayName, createdConversationProfile.getDisplayName());
+    List<SuggestionFeatureConfig> featureConfigsList = createdConversationProfile
+          .getHumanAgentAssistantConfig().getHumanAgentSuggestionConfig().getFeatureConfigsList();
+    assertEquals(2, featureConfigsList.size());
+    assertTrue(featureConfigsList
+          .stream()
+          .filter(featureConfig -> 
+            featureConfig.getSuggestionFeature().getType().equals(Type.ARTICLE_SUGGESTION))
+          .anyMatch(featureConfig -> 
+            KnowledgeBaseName.parse(
+              featureConfig
+                .getQueryConfig().getKnowledgeBaseQuerySource().getKnowledgeBases(0)
+            ).getKnowledgeBase().equals(articleSuggestionKnowledgeBaseId)));
+    assertTrue(featureConfigsList
+          .stream()
+          .filter(featureConfig -> 
+            featureConfig.getSuggestionFeature().getType().equals(Type.FAQ))
+          .anyMatch(featureConfig -> 
+            KnowledgeBaseName.parse(
+              featureConfig
+                .getQueryConfig().getKnowledgeBaseQuerySource().getKnowledgeBases(0)
+            ).getKnowledgeBase().equals(faqKnowledgeBaseId)));
+
+    // Delete the conversation profile
+    String conversationProfileId = ConversationProfileName.parse(
+        createdConversationProfile.getName()).getConversationProfile();
+    try (ConversationProfilesClient conversationProfilesClient = ConversationProfilesClient.create()) {
+      ConversationProfileName conversationProfileName = 
+          ConversationProfileName.ofProjectLocationConversationProfileName(
+            PROJECT_ID, location, conversationProfileId);
+      conversationProfilesClient.deleteConversationProfile(conversationProfileName.toString());
+    }
```

Comment:
You might want to move this to the `tearDown`. It's possible to leak resources if the creation succeeds, but the test assertion fails and thus the deletion never runs.

You could do this by maintaining a resource name to cleanup and in the teardown only delete it if it was set.

Reply:
Good point, thanks